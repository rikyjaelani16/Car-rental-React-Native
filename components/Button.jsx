import { Pressable, Text } from "react-native";
import { useCallback } from "react";
import Animated, { useSharedValue, withSpring } from "react-native-reanimated";

const AnimatedPressable = Animated.createAnimatedComponent(Pressable);

export default Button = ({
  title,
  children,
  color,
  disabled = false,
  onPress,
  style,
}) => {
  const opacity = useSharedValue(1);

  const handlePressIn = () => {
    opacity.value = withSpring(0.8);
  };

  const handlePressOut = () => {
    opacity.value = withSpring(1);
  };

  return (
    <AnimatedPressable
      onPressIn={handlePressIn}
      onPressOut={handlePressOut}
      onPress={onPress}
      disabled={disabled}
      style={{
        opacity: disabled ? 0.5 : opacity,
        backgroundColor: color || "#fff",
        paddingVertical: 8,
        alignItems: "center",
        color: "#fff",
        ...style,
      }}
    >
      {!children ? (
        <Text
          style={{
            color: "#fff",
            fontFamily: "PoppinsBold",
            fontSize: 14,
          }}
        >
          {title}
        </Text>
      ) : (
        children
      )}
    </AnimatedPressable>
  );
};
