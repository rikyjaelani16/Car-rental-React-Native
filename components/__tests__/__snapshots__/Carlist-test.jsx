import * as React from "react";
import renderer from "react-test-renderer";
import Button from "../../Button";

it(`Carlist renders correctly`, () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});
