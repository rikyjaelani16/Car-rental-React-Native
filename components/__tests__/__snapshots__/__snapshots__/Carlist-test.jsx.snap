// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carlist renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderColor": "rgba(0,0,0,0.5)",
      "borderRadius": 2,
      "borderWidth": 0.5,
      "capacityText": {
        "color": "#8A8A8A",
      },
      "carName": {
        "fontSize": 14,
      },
      "card": {
        "borderColor": "rgba(0,0,0,0.5)",
        "borderRadius": 2,
        "borderWidth": 0.5,
        "marginBottom": 20,
        "padding": 20,
      },
      "img": {
        "height": 60,
        "objectFit": "contain",
        "width": 60,
      },
      "marginBottom": 20,
      "padding": 20,
      "price": {
        "color": "#5CB85F",
      },
      "textIcon": {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": "2",
      },
    }
  }
>
  <View
    style={
      {
        "alignItem": "center",
        "flexDirection": "row",
        "gap": 16,
        "justifyContent": "flex-start",
      }
    }
  >
    <View>
      <Image
        style={
          {
            "height": 60,
            "objectFit": "contain",
            "width": 60,
          }
        }
      />
    </View>
    <View>
      <Text
        style={
          {
            "fontSize": 14,
          }
        }
      >
        Innova
      </Text>
      <View
        style={
          {
            "alignItem": "baseline",
            "flexDirection": "row",
            "gap": 16,
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            {
              "alignItem": "baseline",
              "flexDirection": "row",
              "gap": 0,
              "justifyContent": "flex-start",
            }
          }
        >
          <Text />
          <Text
            style={
              {
                "color": "#8A8A8A",
              }
            }
          />
        </View>
        <View>
          <Text>
            <Text />
            <Text
              style={
                {
                  "color": "#8A8A8A",
                }
              }
            >
               
              4
            </Text>
          </Text>
        </View>
      </View>
      <Text
        style={
          {
            "color": "#5CB85F",
          }
        }
      >
        RpÂ 50.000,00
      </Text>
    </View>
  </View>
</View>
`;
